h5py/h5t.pyx

```diff
@@ -1598,19 +1600,25 @@ cpdef TypeID py_create(object dtype_in, bint logical=0, bint aligned=0):
     HDF5 datatype object.  The result is guaranteed to be transient and
     unlocked.
 
-    Argument dtype_in may be a dtype object, or anything which can be
-    converted to a dtype, including strings like '<i4'.
-
-    logical
-        If this flag is set, instead of returning a byte-for-byte identical
-        representation of the type, the function returns the closest logically
-        appropriate HDF5 type.  For example, in the case of a "hinted" dtype
-        of kind "O" representing a string, it would return an HDF5 variable-
+    :param dtype_in: may be a dtype object, or anything which can be
+        converted to a dtype, including strings like '<i4' or an "int".
+    :param logical: when this flag is set, instead of returning a byte-for-byte
+        identical representation of the type, the function returns the closest
+        logically appropriate HDF5 type.  For example, in the case of a "hinted"
+        dtype of kind "O" representing a string, it would return an HDF5 variable-
         length string type.
     """
-    cdef dtype dt = dtype(dtype_in)
-    cdef char kind = dt.kind
+    cdef:
+        cnp.dtype dt
+        char kind
+
+    if isinstance(dtype_in, np.dtype):
+        dt = np.dtype(dtype_in)
+    else:
+        dt = dtype_in = np.dtype(dtype_in)
+    # From now ondtype_in is actually a numpy.dtype
```

Comment:
Sorry to be nitpicky, but... this doesn't really answer my question. Why use `dtype_in.names` as opposed to `dt.names`? In what situation would they be different? Do we have a test that would cover that situation?